api/Controllers/ReportsController.cs

```diff
@@ -373,44 +323,36 @@ private Report CreateInvoice(DateTime time, Template invoiceTemplate, CreateRepo
 					Xlsx = xlsxms.ToArray(),
 					Pdf = pdfms.ToArray()
 				};
+>>>>>>> development
 			}
-
-			return report;
 		}
 
-		private Report CreateInvoice(DateTime time, CreateReport create)
-		{
-			// get template
-			var invoiceTemplate = _templates.Get(create.TemplateId.Value);
-			if (invoiceTemplate == null)
-				throw new MissingTemplateException(create.TemplateId.Value);
-
-			return CreateInvoice(time, invoiceTemplate, create);
-		}
-
-		private Report CreateInvoice(CreateReport create) => CreateInvoice(DateTime.Now, create);
-
 		private Report CreateBulkInvoice(DateTime time, Template invoiceTemplate, CreateReport create)
 		{
-			var reports = _reports.GetMany(
-				type: ReportType.FromString("Invoice"),
-				// year: create.SchoolYear,
-				scope: create.BulkInvoice.Scope,
-				approved: create.BulkInvoice.Approved
-			).ToList();
-
-			var invoices = reports.Select(r => JsonConvert.DeserializeObject<Invoice>(r.Data)).OrderBy(i => i.SchoolDistrict.Name).ToList();
+			var reporter = _reporters.CreateBulkInvoiceReporter(_context);
+			var invoice = reporter.GenerateReport(new BulkInvoiceReporter.Config
+			{
+				SchoolYear = create.SchoolYear,
+				Scope = create.BulkInvoice.Scope,
+				Prepared = time,
+				AsOf = create.BulkInvoice.AsOf,
+				ToSchoolDistrict = create.BulkInvoice.ToSchoolDistrict,
+				ToPDE = create.BulkInvoice.ToPDE,
+				Auns = create.BulkInvoice.Auns,
+				PaymentType = create.BulkInvoice.PaymentType,
+			});
 
 			DateTime toSchoolDate = new DateTime();
 			DateTime toPDEDate = new DateTime();
 
 			// compose workbook
 			var wb = new Workbook(new MemoryStream(invoiceTemplate.Content));
 			InitializeWorkbookSheetPrinterMargins(wb);
-			Console.WriteLine($"ReportsController.CreateBulkInvoice():  number of invoices is {invoices.Count}.");
-			for (int i = 0; i < invoices.Count; i++)
+
+			var districts = invoice.Districts.ToList();
+			for (int i = 0; i < districts.Count; i++)
 			{
-				var invoice = invoices[i];
+				var district = districts[i];
 
 				if (i == 0)
 				{
```

Comment:
Is this supposed to be empty?  Looks like an artifact prior to the changes here.

Reply:
I'm not sure what this comment is in reference to. What's empty?

Reply:
My head, apparently.  Ignore, please.

Reply:
👍